kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -133,15 +132,27 @@ public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
     topicPartitionManager.resumeAll();
   }
 
-  private PartitionedTableId getRecordTable(SinkRecord record) {
-    // Dynamically update topicToBaseTableIds mapping. topicToBaseTableIds was used to be
-    // constructed when connector starts hence new topic configuration needed connector to restart.
-    // Dynamic update shall not require connector restart and shall compute table id in runtime.
-    if (!topicsToBaseTableIds.containsKey(record.topic())) {
-      TopicToTableResolver.updateTopicToTable(config, record.topic(), topicsToBaseTableIds);
+  private void ensureExistingTable(TableId table) {
+    BigQuery bigQuery = getBigQuery();
+    if (bigQuery.getTable(table) == null) {
+      logger.warn("You may want to enable auto table creation by setting {}=true in the properties file",
+              config.TABLE_CREATE_CONFIG);
+      throw new BigQueryConnectException("Table '" + table + "' does not exist");
     }
+  }
 
-    TableId baseTableId = topicsToBaseTableIds.get(record.topic());
+  private PartitionedTableId getRecordTable(SinkRecord record) {
+
+    // SMT RegexTransformation replaces the topic with <Dataset>=<TableName>
+    String dataset = record.topic().split("=")[0];
+    String tableName = record.topic().split("=")[1];
```

Comment:
Is there heavy demand for dynamic routing to different datasets? An alternative here could be to allow the user to configure a single static dataset in the connector and, if multiple datasets are desired, use multiple connector instances.

Or, for something in the middle, we can allow the user to specify a static dataset as a "default" in case the record topic doesn't use the `<dataset>=<table>` syntax.

What do you think?

Reply:
This sounds good to me

Reply:
Will update accordingly